### ISSUE TITLE
Camera FPS

### ISSUE THREAD

**Issue by haroke4** [juliansteenbakker/mobile_scanner#1229]
Hi! Thanks for your plugin!

When I use class `MobileScanner`, the camera view seems to run at 30 (or even lower) fps. Is this because my app runs on the debug version, or is it because of the plugin?
I was using an iPhone 13 with a full battery percentage.
How can I make it so it runs at max fps possible?

The code I was using is `example/lib/barcode_scanner_simple.dart`. [()](https://github.com/juliansteenbakker/mobile_scanner/blob/master/example/lib/barcode_scanner_simple.dart)

**Comment by navaronbracke** [comment]
I think it is intentional that the camera preview runs at 30 FPS, I haven't found documentation on why that is, though.

**Comment by khulegu** [comment]
Yeah, but the old library [qr_code_scanner](https://github.com/juliuscanute/qr_code_scanner) has much better framerate.

**Comment by navaronbracke** [comment]
Apparently this is because running on higher frame rates requires some configuration.

See https://stackoverflow.com/questions/57485050/how-to-increase-frame-rate-with-android-camerax-imageanalysis

**Comment by haroke4** [comment]
Hi, what about iOS? It looks very cheap when camera's fps is low T-T

**Comment by CodeAndThink** [comment]
Hello author, is there any solution to this problem?

**Comment by haroke4** [comment]
Hi @CodeAndThink , unfortunately, I haven't found any solution to fix the low fps issue on this package. However, I found another package called [camerawesome](https://pub.dev/packages/camerawesome), where the FPS is better. You can check it out

**Comment by navaronbracke** [comment]
@CodeAndThink the possible solution mentioned in https://github.com/juliansteenbakker/mobile_scanner/issues/1229#issuecomment-2475627102 should be investigated.
It turns out that CameraX by default uses a lower FPS and this needs to be manually configured.

On iOS/MacOS this also needs to be configured manually in the AVCaptureDevice:
https://stackoverflow.com/questions/14039251/ios-avcapturesession-how-to-get-set-the-number-of-frames-per-second-recorded

@haroke4 I do understand that you want to help out other users here, but I'd rather keep this issue on topic for `mobile_scanner`.

**Comment by superhard** [comment]
The low camera fps is due to this line:
 
```Swift
captureSession!.sessionPreset = AVCaptureSession.Preset.photo
```

I've fixed it in my local fork like this almost a year ago:

```Swift
let presets: [AVCaptureSession.Preset] = [
        .high,
        .photo
    ]

    for preset in presets {
        if captureSession!.canSetSessionPreset(preset) {
            captureSession!.sessionPreset = preset
            break
        }
    }
```

However, it may be sufficient to simply set it to  ```AVCaptureSession.Preset.high```

**Comment by aiserrock** [comment]
@superhard 

Yes. It's enought

captureSession!.sessionPreset = AVCaptureSession.Preset.high

Do the respected developers want to add the ability to influence the frame rate? @navaronbracke @ juliansteenbakker?

**Comment by navaronbracke** [comment]
Yes, we can add this to fix that bug. A PR to do this is also welcome!

**Comment by kevtechi** [comment]
@navaronbracke PR is related to this.
